dart/lib/src/sentry_options.dart

```diff
@@ -327,6 +328,19 @@ class SentryOptions {
   /// The default is 3 seconds.
   Duration? idleTimeout = Duration(seconds: 3);
 
+  final _causeExtractorsByType = <Type, ExceptionCauseExtractor>{};
+
+  Map<Type, ExceptionCauseExtractor> get causeExtractorsByType =>
+      _causeExtractorsByType;
+
+  void addTypedExceptionCauseExtractor(
+      Type type, ExceptionCauseExtractor extractor) {
+    _causeExtractorsByType[type] = extractor;
+  }
```

Comment:
I like this approach.

Reply:
Do you have an opinion on the naming?

Reply:
> Can't we just add the throwable to the `SentryException`?

yeah, I was thinking about this as well, the `throwable` does not need to be serialized/deserialized but can be used in order to enrich the event with extra info, when needed.